### ISSUE TITLE
C3

### ISSUE THREAD

**Issue by yegor256** [yegor256/jpeek#64]
Let's implement C3 cohesion metric:

The Conceptual Cohesion of Classes. Andrian Marcus, Denys Poshyvanyk. Department of Computer Science. Wayne State University. Detroit Michigan 48202.

https://pdfs.semanticscholar.org/c6c4/b63f3ba81f2bee75a0c29367da99971dfbaa.pdf

**Comment by 0crat** [comment]
Job [#64](https://github.com/yegor256/jpeek/issues/64) is now in scope, role is `DEV`

**Comment by 0crat** [comment]
Bug was reported, see [§29](http://datum.zerocracy.com/pages/policy.html#29): +15 points just awarded to [@yegor256](http://www.0crat.com/u/yegor256), total is [+7225](http://www.0crat.com/u/yegor256)

**Comment by 0crat** [comment]
The job [#64](https://github.com/yegor256/jpeek/issues/64) assigned to @icok[/z](http://www.0crat.com/u/icok). The budget is 30 minutes, see [§4](http://datum.zerocracy.com/pages/policy.html#4). Please, read [§8](http://datum.zerocracy.com/pages/policy.html#8) and [§9](http://datum.zerocracy.com/pages/policy.html#9). If the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html).

**Comment by 0crat** [comment]
@icok[/z](https://www.0crat.com/u/icok) this job was assigned to you 5 days ago. It will be taken away from you soon, unless you close it, see [§8](http://datum.zerocracy.com/pages/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by 0crat** [comment]
@icok[/z](https://www.0crat.com/u/icok) this job was assigned to you 8 days ago. It will be taken away from you soon, unless you close it, see [§8](http://datum.zerocracy.com/pages/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by 0crat** [comment]
The user @icok[/z](https://www.0crat.com/u/icok) resigned from [#64](https://github.com/yegor256/jpeek/issues/64), please stop working

**Comment by 0crat** [comment]
The job [#64](https://github.com/yegor256/jpeek/issues/64) assigned to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil). The budget is 30 minutes, see [§4](http://datum.zerocracy.com/pages/policy.html#4). Please, read [§8](http://datum.zerocracy.com/pages/policy.html#8) and [§9](http://datum.zerocracy.com/pages/policy.html#9). If the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html).

**Comment by amihaiemil** [comment]
@0crat waiting for https://github.com/yegor256/jpeek/issues/132

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
java.io.IOException: com.amazonaws.SdkClientException: Unable to execute HTTP request: null
	at com.zerocracy.radars.github.ReSafe.lambda$react$1(ReSafe.java:76)
	at org.cactoos.func.FuncOf.lambda$new$3(FuncOf.java:88)
	at org.cactoos.func.FuncOf.apply(FuncOf.java:104)
	at org.cactoos.func.FuncWithFallback.apply(FuncWithFallback.java:90)
	at org.cactoos.func.IoCheckedFunc.lambda$apply$0(IoCheckedFunc.java:59)
	at org.cactoos.scalar.IoCheckedScalar.value(IoCheckedScalar.java:70)
	at org.cactoos.func.IoCheckedFunc.apply(IoCheckedFunc.java:59)
	at com.zerocracy.radars.github.ReSafe.react(ReSafe.java:80)
	at com.zerocracy.radars.github.ReOnComment.send(ReOnComment.java:206)
	at com.zerocracy.radars.github.ReOnComment.react(ReOnComment.java:123)
	at com.zerocracy.radars.github.ReOnReason.react(ReOnReason.java:58)
	at com.zerocracy.radars.github.ReLogged.react(ReLogged.java:55)
	at com.zerocracy.radars.github.RbOnComment.react(RbOnComment.java:62)
	at com.zerocracy.radars.github.RbByActio...
```

[1.0-SNAPSHOT](https://github.com/zerocracy/farm/tree/1.0-SNAPSHOT) /cc @yegor256

**Comment by 0crat** [comment]
> @0crat waiting for https://github.com/yegor256/jpeek/issues/132 ([here](https://github.com/yegor256/jpeek/issues/64#issuecomment-361848453))

@amihaiemil The impediment for [#64](https://github.com/yegor256/jpeek/issues/64) was registered successfully by @amihaiemil[/z](https://www.0crat.com/u/amihaiemil)

**Comment by amihaiemil** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/yegor256/jpeek/issues/64#issuecomment-361848909))

@amihaiemil This is what I know about this job, as in [§32](http://datum.zerocracy.com/pages/policy.html#32):

  * The job [#64](https://github.com/yegor256/jpeek/issues/64) is [in scope](http://datum.zerocracy.com/pages/policy.html#14) for 17days
  * The role is `DEV`
  * The job is assigned to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil) for [1days](http://datum.zerocracy.com/pages/policy.html#8)
  * There is a monetary reward attached
  * @amihaiemil[/z](https://www.0crat.com/u/amihaiemil) will get 0.50 of their hourly rate for this job
  * The job has an [impediment](http://datum.zerocracy.com/pages/policy.html#9)
  * The [budget](http://datum.zerocracy.com/pages/policy.html#4) is 30 minutes
  * These users are banned and won't be assigned:
    * @yegor256[/z](https://www.0crat.com/u/yegor256): This user reported the ticket
    * @icok[/z](https://www.0crat.com/u/icok): User was resigned from the ticket

**Comment by amihaiemil** [comment]
@0crat refuse, assign someone else

**Comment by 0crat** [comment]
> @0crat refuse, assign someone else ([here](https://github.com/yegor256/jpeek/issues/64#issuecomment-363784366))

@amihaiemil Can't understand "refuse,", try one of these:
  - `assign` Assign a performer to this issue
  - `hello` Just say hello
  - `in` Register this issue in scope (WBS)
  - `out` Remove this issue from scope (WBS)
  - `resign` Resign from this issue
  - `status` Check the status of the job
  - `version` Print current version of the bot
  - `waiting` Register an impediment for a job

**Comment by amihaiemil** [comment]
@0crat resign

**Comment by 0crat** [comment]
> @0crat resign ([here](https://github.com/yegor256/jpeek/issues/64#issuecomment-363784581))

@amihaiemil The user @amihaiemil[/z](https://www.0crat.com/u/amihaiemil) resigned from [#64](https://github.com/yegor256/jpeek/issues/64), please stop working

**Comment by 0crat** [comment]
> @0crat resign ([here](https://github.com/yegor256/jpeek/issues/64#issuecomment-363784581))

@amihaiemil Job `gh:yegor256/jpeek#64` is not assigned, can't get performer

**Comment by 0crat** [comment]
Tasks refusal is discouraged, see [§6](http://datum.zerocracy.com/pages/policy.html#6): -15 points just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), total is [+345](https://www.0crat.com/u/amihaiemil)

**Comment by 0crat** [comment]
Order was successfully finished: +30 points just awarded to @rok-povsic[/z](https://www.0crat.com/u/rok-povsic), total is [+60](https://www.0crat.com/u/rok-povsic)

**Comment by 0crat** [comment]
The job [#64](https://github.com/yegor256/jpeek/issues/64) is now out of scope

**Comment by 0pdd** [comment]
@yegor256 the puzzle [#498](https://github.com/cqfn/jpeek/issues/498) is still not solved.

**Comment by 0crat** [comment]
> @0crat waiting for https://github.com/yegor256/jpeek/issues/132 ([here](https://github.com/cqfn/jpeek/issues/64#issuecomment-361848453))

@amihaiemil Job `gh:cqfn/jpeek#64` is not assigned, can't get performer

**Comment by 0crat** [comment]
> @0crat refuse, assign someone else ([here](https://github.com/cqfn/jpeek/issues/64#issuecomment-363784366))

@amihaiemil Can't understand "refuse,", try one of these:
  - `assign` Assign a performer to this issue
  - `boost` Set a boost for the job
  - `continue` Remove a job's impediment
  - `elect` Trigger election for ticket
  - `hello` Just say hello
  - `in` Register this issue in scope (WBS)
  - `out` Close the order and remove this job from scope
  - `pay` Pay a user some extra cash
  - `quality` Review a task
  - `resign` Resign from this issue
  - `status` Check the status of the job
  - `version` Print current version of the bot
  - `wait` Register an impediment for a job

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/cqfn/jpeek/issues/64#issuecomment-361848909))

@amihaiemil This is what I know about this job in C7JGJ00DP, as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#64](https://github.com/cqfn/jpeek/issues/64) is not in scope
  * Job [footprint](https://www.0crat.com/footprint/C7JGJ00DP?q=%7Bjob%3A%27gh%3Acqfn%2Fjpeek%2364%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by 0crat** [comment]
> @0crat resign ([here](https://github.com/cqfn/jpeek/issues/64#issuecomment-363784581))

@amihaiemil Job `gh:cqfn/jpeek#64` is not assigned, can't get performer

**Comment by 0pdd** [comment]
@yegor256 the puzzle [#576](https://github.com/cqfn/jpeek/issues/576) is still not solved; solved: [#498](https://github.com/cqfn/jpeek/issues/498).
